sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -185,11 +191,22 @@ bool handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
               Opts.find("-cl-std=CL2.0") != string_class::npos;
           const bool RequiresUniformWGSize =
               Opts.find("-cl-uniform-work-group-size") != string_class::npos;
+          std::string LocalWGSize = std::to_string(NDRDesc.LocalSize[0]) +
+                                    ", " +
+                                    std::to_string(NDRDesc.LocalSize[1]) +
+                                    ", " + std::to_string(NDRDesc.LocalSize[2]);
+          std::string GlobalWGSize =
+              std::to_string(NDRDesc.GlobalSize[0]) + ", " +
+              std::to_string(NDRDesc.GlobalSize[1]) + ", " +
+              std::to_string(NDRDesc.GlobalSize[2]);
           std::string message =
               LocalExceedsGlobal
-                  ? "Local workgroup size greater than global range size. "
-                  : "Global_work_size not evenly divisible by "
-                    "local_work_size. ";
+                  ? "Local workgroup size {" + LocalWGSize +
+                        "} is greater than global range size {" + GlobalWGSize +
+                        "}"
+                  : "Global work size {" + GlobalWGSize +
+                        "} is not evenly divisible by localgroup size {" +
```

Comment:
```suggestion
                        "} is not evenly divisible by local work-group size {" +
```

Reply:
Fixed in https://github.com/intel/llvm/pull/2860/commits/e05a86f5e1e37c41bc9d4783d537c51e8a614ba9. Thank you.